flutter-idea/src/io/flutter/analytics/FlutterAnalysisServerListener.java

```diff
@@ -0,0 +1,521 @@
+        pathToErrorTimestamps.put(filePath, nowInstant);
+        pathToHighlightTimestamps.put(filePath, nowInstant);
+        pathToOutlineTimestamps.put(filePath, nowInstant);
+      }
+
+      @Override
+      public void selectionChanged(@NotNull FileEditorManagerEvent event) {
+      }
+
+      @Override
+      public void fileClosed(@NotNull final FileEditorManager source, @NotNull final VirtualFile file) {
+      }
+    };
+    project.getMessageBus().connect().subscribe(FileEditorManagerListener.FILE_EDITOR_MANAGER, fileEditorManagerListener);
+    this.quickFixListener = new QuickFixListener();
+    this.requestListener = new FlutterRequestListener();
+    this.responseListener = new FlutterResponseListener();
+    DartAnalysisServerService analysisServer = DartAnalysisServerService.getInstance(project);
+    analysisServer.setServerLogSubscription(true);
+    analysisServer.addQuickFixListener(this.quickFixListener);
+    analysisServer.addRequestListener(this.requestListener);
+    analysisServer.addResponseListener(this.responseListener);
+  }
+
+  @NotNull
+  public static FlutterAnalysisServerListener getInstance(@NotNull final Project project) {
+    return Objects.requireNonNull(project.getService(FlutterAnalysisServerListener.class));
+  }
+
+  @NotNull
+  private static String safelyGetString(JsonObject jsonObject, String memberName) {
+    if (jsonObject != null && StringUtil.isNotEmpty(memberName)) {
+      JsonElement jsonElement = jsonObject.get(memberName);
+      if (jsonElement != null) {
+        return Objects.requireNonNull(jsonElement.getAsString());
+      }
+    }
+    return "";
+  }
+
+  @Override
+  public void dispose() {
+    // This is deprecated and marked for removal in 2021.3. If it is removed we will
+    // have to do some funny stuff to support older versions of Android Studio.
+    LookupManager.getInstance(project).removePropertyChangeListener(this::onPropertyChange);
+  }
+
+  @Override
+  public void computedAnalyzedFiles(List<String> list) {
+  }
+
+  @Override
+  public void computedAvailableSuggestions(@NotNull List<AvailableSuggestionSet> list, int[] ints) {
+  }
+
+  @Override
+  public void computedCompletion(String completionId,
+                                 int replacementOffset,
+                                 int replacementLength,
+                                 List<CompletionSuggestion> completionSuggestions,
+                                 List<IncludedSuggestionSet> includedSuggestionSets,
+                                 List<String> includedElementKinds,
+                                 List<IncludedSuggestionRelevanceTag> includedSuggestionRelevanceTags,
+                                 boolean isLast,
+                                 String libraryFilePathSD) {
+  }
+
+  @Override
+  public void computedErrors(String path, List<AnalysisError> list) {
+    assert list != null;
+    list.forEach(this::logAnalysisError);
+    pathToErrors.put(path, list);
```

Comment:
what happens if you restart the analysis server? Do we get a new event with new errors.
Wish we had a merged Dart and Flutter plugin so we didn't have to duplicate this logic that the analysis errors window already handles.

Reply:
I added that test and asked for an opinion from jwren.

Reply:
No right choice.   We can discuss, there are pros and cons for both (as with all things).  If you don't send the information you won't have the signal that users are re-analyzing which is a signal in-itself, but if you do send the information there will be fuller logs with the same information.